server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/delivery/SieveMailet.java

```diff
@@ -57,116 +57,130 @@
+        public Builder deliveryHeader(String deliveryHeader) {
+            this.deliveryHeader = deliveryHeader;
+            return this;
         }
-    }
 
-    public ResourceLocator getLocator() {
-        return locator;
-    }
+        public Builder resetReturnPath(boolean resetReturnPath) {
+            this.resetReturnPath = resetReturnPath;
+            return this;
+        }
 
-    public void setLocator(ResourceLocator locator) {
-        this.locator = locator;
-    }
+        public Builder verbose(boolean verbose) {
+            this.verbose = Optional.of(verbose);
+            return this;
+        }
 
-    public Poster getPoster() {
-        return poster;
-    }
+        public Builder consume(boolean consume) {
+            this.consume = consume;
+            return this;
+        }
 
-    public void setPoster(Poster poster) {
-        this.poster = poster;
-    }
+        public Builder quiet(boolean quiet) {
+            this.quiet = Optional.of(quiet);
+            return this;
+        }
 
-    public boolean isConsume() {
-        return consume;
-    }
+        public SieveMailet build() throws MessagingException {
+            if (resourceLocator == null) {
+                throw new MailetException("Not initialised. Please ensure that the mailet container supports either setter or constructor injection");
+            }
+            return new SieveMailet(usersRepos,mailboxManager, resourceLocator, folder, deliveryHeader, resetReturnPath, consume, verbose.or(false), quiet.or(false));
+        }
 
-    public void setConsume(boolean consume) {
-        this.consume = consume;
     }
 
-    public boolean isVerbose() {
-        return verbose;
-    }
+    private final UsersRepository usersRepos;
+    private final MailboxManager mailboxManager;
+    private final String folder;
+    private final ResourceLocator resourceLocator;
+    private final String deliveryHeader;
+    private final boolean resetReturnPath;
+    private final boolean isInfo;
+    private final boolean verbose;
+    private final boolean consume;
+    private final SieveFactory factory;
+    private final ActionDispatcher actionDispatcher;
+    private final Log log;
+
+    private SieveMailet(UsersRepository usersRepos, MailboxManager mailboxManager, ResourceLocator resourceLocator, String folder, String deliveryHeader,
+                       boolean resetReturnPath, boolean consume, boolean verbose, boolean quiet) throws MessagingException {
 
-    public void setVerbose(boolean verbose) {
+        this.usersRepos = usersRepos;
+        this.resourceLocator = resourceLocator;
+        this.mailboxManager = mailboxManager;
+        this.folder = folder;
+        this.actionDispatcher = new ActionDispatcher();
+        this.deliveryHeader = deliveryHeader;
+        this.resetReturnPath = resetReturnPath;
+        this.consume = consume;
+        this.isInfo = verbose || !quiet;
```

Comment:
This variable was later removed as it delivered no value. A log object can now not be quiet and verbose at the same time.